From 17bd7fa80f322bfff0e0fe060a5d20f1127a3fe4 Mon Sep 17 00:00:00 2001
From: Isuru Fernando <isuruf@gmail.com>
Date: Mon, 29 Mar 2021 16:24:46 -0500
Subject: [PATCH] Link pthread and rt at the end

NEO_STATICALLY_LINKED_LIBRARIES_MOCKABLE might need rt
and linking order matters.
---
 shared/test/unit_test/CMakeLists.txt | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/shared/test/unit_test/CMakeLists.txt b/shared/test/unit_test/CMakeLists.txt
index a9963ef2d..e4a136bdd 100644
--- a/shared/test/unit_test/CMakeLists.txt
+++ b/shared/test/unit_test/CMakeLists.txt
@@ -99,18 +99,18 @@ if(NOT SKIP_UNIT_TESTS)
                              ${NEO_SHARED_TEST_DIRECTORY}/common/helpers/includes${BRANCH_DIR_SUFFIX}
   )
 
-  if(UNIX)
-    target_link_libraries(${TARGET_NAME} pthread rt)
-  else()
-    target_link_libraries(${TARGET_NAME} dbghelp)
-  endif()
-
   target_link_libraries(${TARGET_NAME}
                         gmock-gtest
                         ${NEO_STATICALLY_LINKED_LIBRARIES_MOCKABLE}
                         compute_runtime_mockable_extra
   )
 
+  if(UNIX)
+    target_link_libraries(${TARGET_NAME} pthread rt)
+  else()
+    target_link_libraries(${TARGET_NAME} dbghelp)
+  endif()
+
   if(MSVC)
     set_target_properties(${TARGET_NAME} PROPERTIES
                           VS_DEBUGGER_COMMAND_ARGUMENTS "--gtest_filter=* --gtest_catch_exceptions=0 --enable_default_listener --disable_pagefaulting_tests"
